sae_lens/training/training_sae.py

```diff
@@ -184,6 +204,38 @@ def encode_with_hidden_pre(
 
         return feature_acts, hidden_pre_noised
 
+    def encode_with_hidden_pre_gated(
```

Comment:
Once we get to the point we're writing different methods, I do wonder if we should just make different classes.